presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -328,8 +328,15 @@ void PrestoServer::run() {
   }
 
   LOG(INFO) << "STARTUP: Starting all periodic tasks...";
+  std::vector<std::shared_ptr<velox::connector::Connector>> connectors;
+  for (auto connectorId : catalogNames) {
+    connectors.emplace_back(velox::connector::getConnector(connectorId));
+  }
   PeriodicTaskManager periodicTaskManager(
-      driverCPUExecutor(), httpServer_->getExecutor(), taskManager_.get());
+      driverCPUExecutor(),
+      httpServer_->getExecutor(),
+      taskManager_.get(),
+      std::move(connectors));
```

Comment:
I'm thinking  - can PeriodicTaskManager fetch all hive connectors itself, rather than being given by Server?
Since server does not own them and they are singleton, we can remove connectors from the PeriodicTaskManager ctor.

Reply:
Yeah.. That will be ideal.. But that requires changes in velox connector API to list connectors. Then land that PR then this one. Landing stuff nowadays takes a long time. I want to see some signals from the cache counters. Here is what I can do: I can work on that in parallel while landing this. Then come back for a refactor. WDYT